### PR TITLE
[dbt] Add DbtManifestPreparer and DagsterDbtManifestPreparer apidoc

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

This PR updates the docstring for `DbtManifestPreparer` and `DagsterDbtManifestPreparer`.

## How I Tested These Changes

make apidoc-build

### CODE REVIEW

**Comment by maximearmstrong** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22546?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#22546** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22546?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#22537** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22537?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22534** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22534?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22513** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22513?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maximearmstrong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 1]
```diff
@@ -55,7 +66,19 @@ def __init__(
         """
         self._generate_cli_args = generate_cli_args or ["parse", "--quiet"]
 
+    @public
```
Comment (sryza) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:69]: Given the other names used here, should this be named something like `prepare_if_local`? Or `ensure_prepared`?

**Review by sryza — CHANGES_REQUESTED** [review]

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py (hunk 1 above)
Comment (maximearmstrong) [reply python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:69]: `ensure_prepared` would make sense. I think making the change in the [PR updating the API](https://github.com/dagster-io/dagster/pull/22534) would make most sense.

**Review by sryza — CHANGES_REQUESTED** [review]
Do we still want to expose these, given our latest conversation?

**Comment by maximearmstrong** [comment]
> Do we still want to expose these, given our latest conversation?

That's a good point. We should not expose these.

Though, I'm in favor of keeping the new docstring as a reference in the code.

**Comment by maximearmstrong** [comment]
> Do we still want to expose these, given our latest conversation?

@sryza That's a good point. We should not expose these.

Though, I'm in favor of keeping the new docstring as a reference in the code.

**Comment by sryza** [comment]
> Though, I'm in favor of keeping the new docstring as a reference in the code.

I'm also very in favor of that.

**Review by sryza — APPROVED** [review]
Nice

**Pull request merged**